lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -3626,21 +3666,47 @@ status_code_e s1ap_handle_sctp_disconnection(s1ap_state_t* state,
   }
 
   if (reset) {
-    // Check if the UE counters for eNB are equal.
-    // If not, the eNB will never switch to INIT state, particularly in
-    // stateless mode.
-    // Exit the process so that health checker can clean-up all Redis
-    // state and restart all stateless services.
-    AssertFatal(
-        enb_association->nb_ue_associated ==
-            enb_association->ue_id_coll.num_elements,
-        "Num UEs associated with eNB (%u) is more than the UEs with valid "
-        "mme_ue_s1ap_id (%zu). This is a deadlock state potentially caused by "
-        "misbehaving eNB; restarting MME. In stateless mode, health management "
-        "service will eventually detect multiple MME restarts due to this "
-        "deadlock state and force sctpd and hence all services to restart.",
-        enb_association->nb_ue_associated,
-        enb_association->ue_id_coll.num_elements);
+    /* Check if the UE counters for eNB are equal.
```

Comment:
Please put this logic in a function call, so it can be reused during Sctp reset and at S1Setup request